### ISSUE TITLE
Error in nerwork generation

### ISSUE THREAD

**Issue by milnus** [fwhelan/coinfinder#25]
Hi, I love Coinfinder and am exited to take a deeper dive into its outputs.
However, I am experiencing a problem when Coinfinder generates an output network. I get the following error:
```Error: Gexf could not find the alpha name amyA_1~~~amyA_2~~~amyA in the components file.``` 

I use a gene presence absence file in Roary format generated by Panaroo, hence the triple tilde.
substituting all '~~~' does not remove fix the error.
I can also see that _amyA_1~~~amyA_2~~~amyA _ is the first element in a sorted list of alphas, so it seems to choke at the first alpha in network generation.

Any idea why Coinfinder chokes when generating the output network?

**Comment by fwhelan** [comment]
Hi milnus, so glad to hear that you're enjoying coinfinder! I'm sorry that you've run into this issue. I've seen something similar before and the culpit seems to be special characters that various steps along the way don't like in their input. I thought I had caught them all, but apparently haven't (though could you confirm you have a recent version of coinfinder installed to be sure?). I wouldn't worry too much that its choking on the first alpha; unless you see a lot of special characters in your alpha list, we should be able to get the (or most of the!) problem solved by just addressing this one.

When you say substituting the triple tilde doesn't fix the issue, at what stage did you do this substitution? In the Panaroo gene_p_a file? It looks like Gexf is looking for `amyA_1~~~amyA_2~~~amyA` but your alpha list has `_amyA_1~~~amyA_2~~~amyA _`. Do all your alpha's start and end with an underscore? If so, there might be an issue with the exact format of your panaroo input; I've been told it doesn't quite match the format of the roary output in terms of the quotation level- that might be worth double checking with your file.

**Comment by milnus** [comment]
Hi Fiona, Thank you for your response!
I used Coinfinder 1.0.4.
The `_` before and after the `amyA_1~~~amyA_2~~~amyA` in the original issue is my mistake. `amyA_1~~~amyA_2~~~amyA` should have been italic, but there is a space between the name and the last `_`, which messes with the formatting. So there is no `_` in the alpha list.

I tied to substitute the `~~~` with `;` in the gene_p_a file, which is given as input to Coinfinder.
I also quoted everything in the gene_p_a file using the `sed` command that you have provided here on Github. (If the gene_p_a file is not quoted Coinfinder throws an error when looking at the tree using R. 😉  )

**Comment by fwhelan** [comment]
I've pushed a bit of code that will help coinfinder fail more gracefully if the gene_p_a file is not quoted as required (commit 8df33eb), and also to treat ~s as special characters (commit a4d6390). In a quick local test, I was able to recreate your issue by adding ~s to a gene cluster ID in the gene_p_a input, which then was solved with this bug fix. Hopefully this will solve your issue, but please let me know if it persists. If you're using coinfinder via bioconda, let me know so I can push a new version for you that will include this fix.

Thanks for your patience!

**Comment by milnus** [comment]
Sorry for the late response, PhD deadlines got to me.
It would be nice with a push to bioconda, as I'm using it.
Cheers!

**Comment by fwhelan** [comment]
No worries! I just created a new release (`1.0.6`) which should be pushed to bioconda automatically. I'll check back in the next  ~24hrs to be sure that's happened appropriately.

Good luck with the PhD deadlines!

**Comment by fwhelan** [comment]
`v1.0.6` of coinfinder should now be up on bioconda for unix and mac os. Let me know if this issue persists with the new version!

**Comment by milnus** [comment]
It is working now! :-D
Thank you!
Do you mind if I ask if you have tried reading the graph into R?
read.gexf() from the package rgexf gives an error related to colours. If you have not tried reading the network into R it's all god!

**Comment by fwhelan** [comment]
Great, glad to hear that it's working ok!

Sorry, no- I've only opened the .gexfs in Gephi myself. If it maybe helps at all, I used this format guide when I was building the code behind generating the .gexf file: https://gephi.org/gexf/format/. If you figure out how to make the file compliant with R's rgexf let me know and I'd be happy to make the change in coinfinder!

**Comment by milnus** [comment]
The problem seems to be `read.gexf()`'s parsing of the hexadecimal colours used for nodes. Coinfinders formatting of the gexf file seems just fine, as I was able to tweak the `read.gexf()` function to read an example graph with hex colours, that produced an error in the current version of `read.gexf()`. I have lunched an issue on the rgexf repo. (https://github.com/gvegayon/rgexf/issues/41#issue-713358961)

Thank you so much for you patience!
